qemu/accel/tcg/translate-all.c

```diff
@@ -677,9 +676,7 @@ page_collection_lock(struct uc_struct *uc, tb_page_addr_t start, tb_page_addr_t
             continue;
         }
         if (page_trylock_add(uc, set, index << TARGET_PAGE_BITS)) {
```

Comment:
But you also remove these function calls, which have side effects obviously.

Reply:
I think `page_trylock_add` also does nothing other than allocating memories. or did I miss anything?

Reply:
Another side effect is `page_entry_destroy` call when destroying objects, but it's also same. `page_lock` is no-op in Unicorn, so I think it's a meaningless call.

Reply:
`page_collection_lock` is used by `tb_invalidate_phys_range`, which is not non-ops in my experience.

Reply:
`page_trylock_add` also inserts new pages, no?

Reply:
hmm interesting. I'm running it without `page_collection_lock`/`unlock`, but no problem yet. `tb_invalidate_phys_range` uses the page collection only for `page_collection_unlock` call.

`page_trylock_add` does not insert new pages. It creates a page entry and insert it to the tree, but it's not used anywhere. I guess it's only for managing order of locks.